tasks/volumes.yml

```diff
@@ -27,7 +27,7 @@
     {% endif %}
     -a {{ ansible_check_mode }}
   with_items: "{{ volumes }}"
-  when: item.type | default(libvirt_volume_default_type) in ['volume', 'network']
+  when: item.type | default(libvirt_volume_default_type) == 'volume'
```

Comment:
@goetzk this condition jumps out at me, given that you added network in https://github.com/stackhpc/ansible-role-libvirt-vm/pull/48/files. This condition sees a lot of changes. Perhaps different people have a different understanding of what the network type means, and we need to include some other condition as well?

Reply:
If I'm not wrong, the script acts on libvirt volume pools only, so it is useless for volume types other than `volume`.

Reply:
Did you try it and see it fail? One thing it does is upload an image to the volume if specified, which could be useful for network volumes.

Reply:
The command is `virsh vol-upload --pool "$POOL" --vol "$NAME" --file "$IMAGE"` so you still need a volume pool. So it's useless with type `network`.

Reply:
Would you not use type=network with an [RBD volume pool](https://github.com/stackhpc/ansible-role-libvirt-host/pull/36)?

Reply:
Yes, but this script is non effective with something else than `volume`. This script is unsuitable for `network` disks.

Reply:
Ok, I just want to be sure because this condition has changed back and forth multiple times.

Reply:
OK then I won't touch it again.

Reply:
Just to be clear: In theory it would be possible in libvirt to refer to a RBD volume pool in a domain definition. But for now, you can define a RBD volume pool, but you can't use it from the VMs. See [the issue I opened about it on libvirt repo](https://gitlab.com/libvirt/libvirt/-/issues/43). So because of that, we can't declare RBD volumes ad `type=volume` because the `volume` refers to a libvirt volume pool.

When the libvirt issue will be solved, it will simplify things a lot, and in most of cases RBD volumes will be declared as `type=volume`.

Reply:
Well in that case, is it useful to add RBD volume pool support, or just confusing?

Reply:
I think it is useful for at least 2 reasons:

- prepare for the future
- AFAIC my provisioning script use `virt vol-create-as` which needs a volume pool.

Reply:
Does that mean that the libvirt-host patch has not been fully tested?

Reply:
What do you mean? I used it already in a production context.

Reply:
Then maybe I don't understand what works and what doesn't. How are you using the RBD volume pool?

Reply:
OK, I will try to recap. I'll say generalities that you know already but this is just to make sure we speak about the same thing, or maybe I am the one not understanding something and you can tell me what.

When defining a domain on libvirt, you have basically two possibilities to add a disk:

- You can refer to a volume which is part of a volume pool, defined in libvirt with the role libvirt-host for example. In that case, the disk will have the type `volume`. The advantage is that the disk declaration in the domain XML is really simple because all the common details have been already given in the volume pool definition. For example, in the case of a LVM volume pool, you don't have to give the name of the volume group again.
- You can use a volume that is not part of a volume pool. In that case you have to use types like `file`, `block`, `dir`, `network`... In that case you have to specify everything you need to access the disk. In the case is RBD, you have to specify hosts, secret name, username, etc...

For now, RBD volume pools can be defined but you cannot refer to them in a domain definition. So, in order to use a RBD disk in a domain you have to use the `network` type and define everything.

So, for a VM to work with a RBD disk, there is no need to define a RBD volume pool.

BUT there are still uses for defining a RBD volume pool. In my case I use them to manage volumes directly from libvirt.

Reply:
Thanks for the summary. I spend more time working with OpenStack than libvirt directly, so not intimately familiar with it. My main concern is that the RBD volume pool support was added without the ability to test it. If you are using it then we can accept it, although it's a shame that it's not too useful.

Reply:
> My main concern is that the RBD volume pool support was added without the ability to test it.

You mean RBD volume pool support in libvirt?

Reply:
No, in the libvirt-hosts role. But you have confirmed that you are using it.